server/src/main/java/org/elasticsearch/index/search/QueryParserHelper.java

```diff
@@ -132,6 +132,10 @@ private QueryParserHelper() {}
                 // Ignore metadata fields
                 continue;
             }
+            if (acceptMetadataField == false && fieldType.isIndexed() == false) {
```

Comment:
The `acceptMetadataField` variable name is highly confusing if we're using it here to denote `full` vs `partial` wilcard. The javadocs for `acceptMetadataField` say "Whether metadata fields should be added when a pattern is expanded.". Non-indexed fields shouldn't be added though, independently of whether they're metadata fields or not? Which metadata fields exist that are not indexed? I find this code now highly confusing, as the relation between `indexed` and `acceptMetadataField` isn't clear, at least with how things are currently named.